src/Symfony/Component/HtmlSanitizer/HtmlSanitizerInterface.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HtmlSanitizer;
+
+/**
+ * Provides an object-oriented API to sanitize untrusted HTML input for safe
+ * insertion into a document’s DOM.
+ *
+ * This interface is inspired by the W3C Standard Draft about a HTML Sanitizer API
+ * ({@see https://wicg.github.io/sanitizer-api/}).
+ *
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * @experimental
+ */
+interface HtmlSanitizerInterface
```

Comment:
needed already?

Reply:
What do you mean? It'll be necessary when injecting the service won't it?

Reply:
i could inject `HtmlSanitizer`, i feel like a custom parser should be the main extension/testing entrypoint at first.

Reply:
Why would this be a better design, from a SOLID pov?

Reply:
it's not. Just questioning the introduction of an experimental interface.

Maybe we can extract better API over time without being stuck on something.

Reply:
from https://symfony.com/doc/current/contributing/code/experimental.html

> gather feedback from real-world usage, adapt the API, or remove it altogether

i'd say

_gather feedback from real-world usage, adapt the API, or add one_ :)

Reply:
Oh OK!
I think that while it's experimental, we should still try to achieve something that is as close as possible to what we would consider stable.
Experimental is here to give a simple path when we realize we make some mistakes along this path.
With this approach, adding the interface makes sense to me.
The corollary here is of course that we should review the PR (and this interface) as if it were not experimental.

Reply:
my point is, we can always extraxct and add an interface, but not always remove one.

So why pushing the interface now, when technically we can do without still (IMHO, or change my mind ;))

Reply:
We can remove it, because experimental. Yet I don't think we will, because the review will confirm we need it I believe. That's why we should add it.

Reply:
let me rephrase it, after the experimental flag is gone, removing is more hassle than adding.

For now im not convinced this interface is needed, but it's not a blocker either.

Reply:
My personal POV is relative usage: if we don't add the interface early on, people will typehint the class implementation, meaning we will need to provide a autowirable service until 7.0. I don't think that's wise: we should always promote the usage of interfaces in typehints IMO, esp when they are linked to BC.

And if we consider the autowirable service experimental as well (allowing us to remove it earlier than 7.0), then we don't have any reason not to use an interface I think.

Reply:
this is unrelated to autowiring. HtmlSanitizer is as autowire-able as HtmlSanitizerInterface.

typehinting interfaces over classes is not a rule by law ;)

the only reason for an interface, technically spoken, is if we expect some form of polymorhism. But generally i believe it's not (eg. in tests you can actually just sanitize HTML isnt it).

Reply:
Software design should allow for what is *not* expected (aka we should not have to anticipate future need for the present design to be correct.) We have SOLID engineering rules on the topic that remind us that this is achieved by type-hinting against abstractions (=interfaces in PHP, for services at least).
That's why the interface makes sense to me. I'm for keeping it, and autowiring it also.

Reply:
sound reasonable :+1: (still not convinced though :))

Let's settle.

Reply:
Well, if we want to introduce a TraceableSanitizer in the future to hook into the debug tools, the interface is a must-have to allow us to implement that through decoration rather than inheritance. If we initially push projects to typehint the class, we cannot use decoration on the service.

Reply:
i'd be curious how much sense a system-wide HtmlSanitizerConfig makes anyway :/